spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/extensions/parameters/ParametersTransformationService.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2016-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.appbroker.extensions.parameters;
+
+import org.springframework.cloud.appbroker.deployer.BackingApplication;
+import org.springframework.cloud.appbroker.deployer.BackingApplications;
+import org.springframework.cloud.servicebroker.exception.ServiceBrokerException;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ParametersTransformationService {
+	private Map<String, ParametersTransformer> parametersTransformersByName = new HashMap<>();
+
+	public ParametersTransformationService(List<ParametersTransformer> parametersTransformers) {
+		parametersTransformers.forEach(parametersTransformer ->
+			this.parametersTransformersByName.put(parametersTransformer.getName(), parametersTransformer));
+	}
+
+	public Mono<BackingApplications> transformParameters(BackingApplications backingApplications,
+														 Map<String, Object> parameters) {
+		return Flux.fromIterable(backingApplications)
+			.flatMap(backingApplication ->
+				Flux.fromIterable(getTransformers(backingApplication))
+					.flatMap(transformerName ->
+						transform(transformerName, backingApplication, parameters)
+					))
+			.then(Mono.just(backingApplications));
```

Comment:
I have a slight concern about this. It might be preferable to collect the backing apps and return that collection instead of the original reference to the `backingApplications` object. I think it should be ok though if it's not operating on a different thread.

Reply:
Returning a `Mono<List<BackingApplication>>` instead of `Mono<BackingApplications>` is problematic. That would mean part of the flow deals with `BackingApplications` and the rest of the flow deals with `List<BackingApplication>`. That inconsistency is going to be confusing. 

We might be able to start the flow with `List<BackingApplication>` and be consistent with that throughout. Then `BackingApplications` would only be used for the Boot properties mapping. I can try that out.